hyperspy/model.py

```diff
@@ -924,475 +956,698 @@ def red_chisq(self):
             ' reduced chi-squared'
         return tmp
 
-    def fit(self, fitter="leastsq", method='ls', grad=False,
-            bounded=False, ext_bounding=False, update_plot=False,
-            **kwargs):
+    def _calculate_parameter_std(self, pcov, cost, ysize):
+        warn_cov = False
+
+        if pcov is None:  # Indeterminate covariance
+            p_var = np.zeros(len(self.p0), dtype=float)
+            p_var.fill(np.inf)
+            warn_cov = True
+        elif isinstance(pcov, np.ndarray):
+            p_var = np.diag(pcov).astype(float) if pcov.ndim > 1 else pcov.astype(float)
+
+            if p_var.min() < 0 or np.any(np.isnan(p_var)) or np.any(np.isinf(p_var)):
+                # Numerical overflow on diagonal
+                p_var.fill(np.inf)
+                warn_cov = True
+            elif ysize > self.p0.size:
+                p_var *= cost / (ysize - self.p0.size)
+            else:
+                p_var.fill(np.inf)
+                warn_cov = True
+        else:
+            raise ValueError(f"pcov should be None or np.ndarray, got {type(pcov)}")
+
+        if warn_cov:
+            _logger.warning(
+                "Covariance of the parameters could not be estimated. "
+                "Estimated parameter standard deviations will be np.inf. "
+                "This could indicate that the model is a poor description "
```

Comment:
Is that always true? If not, I suggest to remove it not to worry the users unnecessarily.

Reply:
Well, the use of "could" qualifies it, but I agree, it could be an unnecessary warning in that sense - removed.